### PR TITLE
Nodelocal updates and CSV wildcard

### LINKED ISSUE #5904: importccl: Allow wildcards in import URIs
PR: https://github.com/cockroachdb/cockroach/pull/40714

From release notes:
> [`IMPORT`](../v20.1/import.html) statements now support specifying CSV filenames using wildcard characters. This behavior can be disabled with the `WITH disabled_glob_matching` option. [#40714][#40714] {% comment %}doc{% endcomment %}

### LINKED ISSUE #5905: storage: Support nodelocal file access using NodeID
PR: https://github.com/cockroachdb/cockroach/pull/41990

From release notes:
> When using a `nodelocal` file URL for [`IMPORT`](../v20.1/import.html)/[`BACKUP`](../v20.1/backup.html)/[`RESTORE`](../v20.1/restore.html)/[`EXPORT`](../v20.1/export.html), you can now specify which node's local file system to use by including the node's ID in the URL: `nodelocal://<nodeID>/path/file.csv`. [#41990][#41990] {% comment %}doc{% endcomment %}

### LINKED ISSUE #6083: Create IMPORT/BACKUP examples without cloud storage
A user on our community Slack:

> I'm looking at the docs about importing a database from an existing database dump from another database type, what's the easiest way without cloud storage and just a few containers? I'm just trying to test an import. I looked at the nodelocal stuff but not sure I fully understand how that works or whether it's what I need

My response: 

> we really need to improve our end-to-end IMPORT guidance for different scenarios, including using local storage. Have you seen the options here https://www.cockroachlabs.com/docs/stable/import.html#import-file-location
>
> Option 1, using a local file server, might be easiest: https://www.cockroachlabs.com/docs/stable/create-a-file-server.html. Unfortunately, we don’t have explicit instructions on leveraging that for IMPORT. I’ll make sure we get that docs work on the backlog. 

This comes up fairly regularly, I think, for import, backup, and restore testing. 

Related 20.1 roadmap items:
- https://github.com/cockroachdb/docs/issues/6010
- https://github.com/cockroachdb/docs/issues/5905

Related docs improvement requests:
- https://github.com/cockroachdb/docs/issues/3724
- https://github.com/cockroachdb/docs/issues/4814

### PR DESCRIPTION
**Pull request by lnhsingh**
- Edit note about nodelocal usage (Closes #5905)
- Add nodelocal IMPORT example (Closes #6083)
- Allow wildcards in IMPORT URLs (Closes #5904)

### CODE REVIEW

File: v20.1/import.md [hunk 1]
```diff
@@ -309,6 +311,59 @@ CSV DATA (
...
 ~~~
 
+#### Using a wildcard
+
+You can specify file patterns to match instead of explicitly listing every file. Use the `*` wildcard character to include matching files directly under the specified path. A wildcard can be used to include:
+
+- All files in a given directory (e.g.,`s3://bucket-name/path/to/data/*`)
+- All files in a given directory that end with a given string (e.g., `s3://bucket-name/files/*.csv`)
+- All files in a given directory that start with a given string (e.g., `s3://bucket-name/files/data*`)
+-All files in a given directory that start and end with a given string (e.g., `s3://bucket-name/files/data*.csv`)
```
Comment (dt) [inline v20.1/import.md]: nit: missing space.

File: v20.1/import.md [hunk 2]
```diff
@@ -309,6 +311,59 @@ CSV DATA (
...
+
```
Comment (dt) [inline v20.1/import.md]: paths are matched using https://golang.org/pkg/path/filepath/#Match if we want to link to it for more details.

File: v20.1/import.md [hunk 3]
```diff
@@ -796,6 +851,46 @@ Google Cloud:
...
+~~~
+
+~~~
+$ cd extern
+$ ls
+
+customers.csv
+~~~
+
+Then, specify the node to access by including the `nodeID` in the `IMPORT` statement:
```
Comment (Amruta-Ranade) [inline v20.1/import.md]: nit: "to access" > "to be accessed"?

File: v20.1/import.md [hunk 4]
```diff
@@ -309,6 +311,59 @@ CSV DATA (
...
+#### Using a wildcard
+
+You can specify file patterns to match instead of explicitly listing every file. Use the `*` wildcard character to include matching files directly under the specified path. A wildcard can be used to include:
+
+- All files in a given directory (e.g.,`s3://bucket-name/path/to/data/*`)
+- All files in a given directory that end with a given string (e.g., `s3://bucket-name/files/*.csv`)
+- All files in a given directory that start with a given string (e.g., `s3://bucket-name/files/data*`)
+-All files in a given directory that start and end with a given string (e.g., `s3://bucket-name/files/data*.csv`)
+
+These only match files directly under the specified path, and do not descend into additional directories recursively.
```
Comment (Amruta-Ranade) [inline v20.1/import.md]: nit: Not sure you need a comma here

File: v19.2/import.md [hunk 5]
```diff
@@ -356,6 +358,59 @@ WITH
...
+#### Using a wildcard
+
+- <span class="version-tag">New in v19.2:</span> You can specify file patterns to match instead of explicitly listing every file. Use the `*` wildcard character to include matching files directly under the specified path. A wildcard can be used to include:
+
+- All files in a given directory (e.g.,`s3://bucket-name/path/to/data/*`)
+- All files in a given directory that end with a given string (e.g., `s3://bucket-name/files/*.csv`)
+- All files in a given directory that start with a given string (e.g., `s3://bucket-name/files/data*`)
+-All files in a given directory that start and end with a given string (e.g., `s3://bucket-name/files/data*.csv`)
+
+These only match files directly under the specified path, and do not descend into additional directories recursively.
```
Comment (Amruta-Ranade) [inline v19.2/import.md]: nit: Not sure you need a comma here

File: _includes/v20.1/misc/external-urls.md [hunk 6]
```diff
@@ -33,18 +29,18 @@ If your environment requires an HTTP or HTTPS proxy server for outgoing connecti
...
 
-- <sup>5</sup>  The host component of NFS/Local can either be empty or the `nodeID`. If the `nodeID` is specified, it is currently ignored (i.e., any node can be sent work and it will look in its local input/output directory); however, the `nodeID` will likely be required in the future.
+- <sup>5</sup>  The host component of NFS/Local can either be empty or the `nodeID`. If you do not specify a `nodeID` when using `nodelocal` storage, the individual data files will be in the `extern` directories of arbitrary nodes and will likely not work as intended. To work correctly, each node must have the [`--external-io-dir` flag](cockroach-start.html#general) point to the same NFS mount or other network-backed, shared storage. <span class="version-tag">New in v20.1:</span> If a `nodeID` is provided, the data files will be in the `extern` directory of the specified node. The `nodeID` will likely be required in the future.
```
Comment (Amruta-Ranade) [inline _includes/v20.1/misc/external-urls.md]: The last sentence is confusing to me: "`nodeID` will likely be required in the future". Does it mean that `nodeID` is optional right now and might be mandatory in the future and so the users should use it now to future-proof their setup? Or will it be required for something else entirely?

**Pull request merged**